src/Discord.Net.Commands/CommandService.cs

```diff
@@ -341,17 +350,55 @@ public void AddTypeReader<T>(TypeReader reader)
         ///     type.
         ///     If <paramref name="type" /> is a <see cref="ValueType" />, a nullable <see cref="TypeReader" /> for the
         ///     value type will also be added.
-        ///     If a default <see cref="TypeReader" /> exists for <paramref name="type" />, a warning will be logged and
-        ///     the default <see cref="TypeReader" /> will be replaced.
         /// </summary>
         /// <param name="type">A <see cref="Type" /> instance for the type to be read.</param>
         /// <param name="reader">An instance of the <see cref="TypeReader" /> to be added.</param>
         public void AddTypeReader(Type type, TypeReader reader)
         {
-            if (_defaultTypeReaders.ContainsKey(type))
-                _ = _cmdLogger.WarningAsync($"The default TypeReader for {type.FullName} was replaced by {reader.GetType().FullName}." +
-                    "To suppress this message, use AddTypeReader<T>(reader, true).");
-            AddTypeReader(type, reader, true);
+            var readers = _typeReaders.GetOrAdd(type, x => new ConcurrentDictionary<Type, TypeReader>());
+            readers[reader.GetType()] = reader;
+
+            if (type.GetTypeInfo().IsValueType)
+                AddNullableTypeReader(type, reader);
+        }
+        /// <summary>
+        ///     Adds a custom entity <see cref="TypeReader" /> to this <see cref="CommandService" /> for the supplied
+        ///     object type.
+        /// </summary>
+        /// <example>
+        ///     <para>The following example adds a custom entity reader to this <see cref="CommandService"/>.</para>
+        ///     <code language="cs" region="AddEntityTypeReader"
+        ///           source="..\..\..\Discord.Net.Examples\Commands\CommandService.Examples.cs" />
+        /// </example>
+        /// <typeparam name="T">The object type to be read by the <see cref="TypeReader"/>.</typeparam>
+        /// <param name="typeReaderGenericType">
+        ///     A <see cref="Type" /> that is a generic type definition with a single open argument
```

Comment:
Sounds a bit weird 🤔

Reply:
Changed it to `A generic type definition (with one open argument) of the <see cref="TypeReader" />.`. Sounds better?